jwst/charge_migration/charge_migration_step.py

```diff
@@ -28,19 +29,21 @@ class ChargeMigrationStep(Step):
     def process(self, input):
 
         # Open the input data model
-        with datamodels.RampModel(input) as input_model:
```

Comment:
Any idea why this was using `RampModel(input)` instead of `datamodels.open(input)`? Are there times when this input might not already be a `RampModel`?

Reply:
There are several other steps modified in this PR that are similar (and the same questions apply).

Reply:
I believe you can use a RampModel up to the ramp step, but if we use the use_datamodel function then it does not matter anymore since it uses datamodels.open()

Reply:
@braingram The calwebb_detector1 pipeline, as well as its individual steps, must have original 4-D data ramps to work with, which is a `RampModel`. Forcing the calwebb_detector1 pipeline module to load its original input into a `RampModel` (as opposed to using a generic open and letting datamodels decide the appropriate model based on the `DATAMODL` keyword) is necessary, because the "uncal" files created by level-1b processing are in the form of a `Level1bModel`. So we have to force it into a `RampModel`. Whether it's also necessary to do that for individual steps, like `charge_migration`, is debatable. In normal flow, as part of the calwebb_detector1 pipeline, the input would already be in the form of a `RampModel`, thanks to calwebb_detector1 focing that. But there's also the possibility of someone trying to run an individual step on an "uncal" file, outside of the pipeline. Hence it could be argued that again forcing it `RampModel` is safest.

Reply:
Thanks! So the steps (which may receive a `Level1bModel` or `RampModel`) could check the model input type (`isinstance(model, RampModel)`) and only convert it to a `RampModel` if it's not already one.

Reply:
this is what the function does now